src/Atmos/Model/source.jl

```diff
@@ -173,3 +181,73 @@ function atmos_source!(
         source.ρu -= β_sponge * (state.ρu .- state.ρ * s.u_relaxation)
     end
 end
+
+"""
+    RemovePrecipitation{FT} <: Source
+
+`q_tot` removal when cloud condensate is exceeding a threshold.
+The threshold is defined either in terms of condensate or supersaturation.
+The removal rate is implemented as a relaxation term
+in the Microphysics_0M module.
+The default thresholds and timescale are defined in CLIMAParameters.jl.
+"""
+struct RemovePrecipitation <: Source
+    " Set to true if using qc based threshold"
+    use_qc_thr::Bool
+end
+function atmos_source!(
+    s::RemovePrecipitation,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    # TODO - should I be using aux here? Or do another saturation adjustement?
+    # TODO - how to assert that it is used with EquilMoist module?
+    #        (if it were to be used with non-EquilMoist model
+    #         q_liq and q_ice would be part of state)
+    FT = eltype(state)
+    if aux.moisture.q_liq + aux.moisture.q_ice > eps(FT)
+
+        q = PhasePartition(
+            state.moisture.ρq_tot / state.ρ,
+            aux.moisture.q_liq,
+            aux.moisture.q_ice,
+        )
+        T::FT = aux.moisture.temperature
+        q_dry::FT = 1 - q.tot
+
+        dqt_dt::FT = 0
+        if s.use_qc_thr
+            dqt_dt = remove_precipitation(atmos.param_set, q)
+        else
+            ts_neq = TemperatureSHumNonEquil(atmos.param_set, T, state.ρ, q)
+            q_vap_sat = q_vap_saturation(ts_neq)
+            dqt_dt = remove_precipitation(atmos.param_set, q, q_vap_sat)
+        end
+
+        _e_int_i0::FT = e_int_i0(atmos.param_set)
+        _cv_d::FT = cv_d(atmos.param_set)
+        _cv_l::FT = cv_l(atmos.param_set)
+        _cv_i::FT = cv_i(atmos.param_set)
+        _T_0::FT = T_0(atmos.param_set)
+
+        source.moisture.ρq_tot += state.ρ / q_dry * dqt_dt
+
+        source.ρ += state.ρ / q_dry * dqt_dt
```

Comment:
```suggestion
        source.ρ += source.moisture.ρq_tot
```

Reply:
My preference would be to collect all mass sources owing moisture sources and add them as a density source where we specify the equations of motion, not piece-by-piece where we define parameterizations.

Reply:
I have opened an issue. I'm no sure how to do it with the current piece-by-piece source term summation. 

issue https://github.com/CliMA/ClimateMachine.jl/issues/1466